### PR TITLE
MicroOS: enable container tests

### PR DESCRIPTION
**Pull request by jlausuch**
Docker is not supported in SUSE MicroOS and Podman comes preinstalled
by default, so we should only run Podman tests.

- Related ticket: https://progress.opensuse.org/issues/76915
- Verification run: http://fromm.arch.suse.de/tests/20

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by Vogtinator** [comment]
The change itself looks good to me, but I'm wondering whether the purpose is still the same.

`get_suse_container_urls` returns names of container images in the current product. SUSE MicroOS doesn't have any relevant ones, so just not returning anything here would also be fine. Currently it ends up running untested images.

If this is used to test the container runtime on MicroOS, it should use the tested and stable ones from `registry.suse.com` instead.

**Comment by jlausuch** [comment]
> `get_suse_container_urls` returns names of container images in the current product. SUSE MicroOS doesn't have any relevant ones, so just not returning anything here would also be fine. Currently it ends up running untested images.

This is right, and SLE container image is already tested by other modules for all versions. I will try to get more clearance in the requirements to what exactly is mean to be tested (who knows? maybe we don't need to test SLE container at all)

**Comment by jlausuch** [comment]
I requested some feedback about this, but it seems Jiri is on vacation and I'm not sure we will get an answer after new's year. What about merging this as it is and do modifications later according to the feedback and requirements?

**Comment by Vogtinator** [comment]
As a compromise, you could mention `registry.suse.com/suse/sle15:15.2` in both `@image_names` and `@stable_names` for now.

The IMO best way would be to just test some SLE resp. openSUSE images in a test like `containers_3rd_party`, just with some more SUSE specific stuff like `zypper` calls on top.

**Comment by jlausuch** [comment]
> As a compromise, you could mention `registry.suse.com/suse/sle15:15.2` in both `@image_names` and `@stable_names` for now.
> 
> The IMO best way would be to just test some SLE resp. openSUSE images in a test like `containers_3rd_party`, just with some more SUSE specific stuff like `zypper` calls on top.

I have decided to remove (for now) SLE based containers, as there is some confusion in requirements. I will enable them (if needed at all) when we get more clearance from PMs (hopefully next year :D).

**Comment by jlausuch** [comment]
http://fromm.arch.suse.de/tests/40

File: lib/containers/container_images.pm
Comment (pdostal) [inline lib/containers/container_images.pm:122]: Here we're checking the container image but `is_microos` is checking the host. Is this really right?

**Review by pdostal — CHANGES_REQUESTED** [review]
I'm sorry to block this. I have just one question and then we can merge this.

File: lib/containers/container_images.pm
Comment (jlausuch) [reply lib/containers/container_images.pm:122]: This function checks that the container version is the same as the host, checking for same string `"SUSE Linux Enterprise Server 15 SPX`. For MicroOS testing SLE containers it doesn't make sense since the host will never show `SUSE Linux Enterprise Server 1...` but `SUSE MicroOS ...`. That's why I skip that check for MicroOS case.

Anyway, I haven't enabled the test case eventually. Maybe I can remove this part of the code or just wait for feedback from Release Manager.

Comment (jlausuch) [reply lib/containers/container_images.pm:122]: see the difference between host and container:
http://fromm.arch.suse.de/tests/18#step/podman_image/97
http://fromm.arch.suse.de/tests/18#step/podman_image/98

Comment (pdostal) [reply lib/containers/container_images.pm:122]: But the `is_microos()` checks the host and the `validate_script_output()` inside the condition checks the container. Why do we check the container only if host isn't MicroOS?

Comment (jlausuch) [reply lib/containers/container_images.pm:122]: Because the script output validation will always fail in MicroOS case, as we have `SUSE Linux Enterprise Server 15 SP2` in the image and something completely different in the host.

Comment (jlausuch) [reply lib/containers/container_images.pm]: I've removed this part as I have disabled SLE image tests.

**Comment by jlausuch** [comment]
New VR: http://fromm.arch.suse.de/tests/42#

**Pull request merged**
